src/docker-osx-dev

```diff
@@ -846,6 +853,21 @@ function initial_sync {
   log_debug "Creating parent directories in Docker VM: $ssh_cmd"
   $DOCKER_HOST_SSH_COMMAND "$ssh_cmd"
 
+  local readonly excludes=($EXCLUDES)
+  exclude_flags=${excludes[@]/#/--exclude }
+  local readonly includes=($INCLUDES)
+  include_flags=${includes[@]/#/--include }
+  for key in "${!parent_dirs[@]}"; do
+    parent_dir=${parent_dirs[$key]}
+    base_name=${base_names[$key]}
+    if $DOCKER_HOST_SSH_COMMAND "test -e '$parent_dir/$base_name'" > /dev/null 2>&1; then
+      log_debug "skipped tar for $parent_dir/$base_name"
+    else
+      log_info "Initial sync using tar for $parent_dir/$base_name"
+      tar -cC "$parent_dir" $include_flags $exclude_flags "$base_name" | $DOCKER_HOST_SSH_COMMAND "tar -xC '$parent_dir'"
```

Comment:
Yes, that's exactly what it does. I experimented with using the tar -z flag as well, but that's a lot slower, because the data transfer is not the bottleneck (it is local traffic after all), and the cpu needs to compress and decompress at the same time. (one on the host, one in the vm)

I haven't tested transferring the tar as a file, but I can't imagine how that could be faster (and I have no time to test it at the moment)